### PR TITLE
12 implement command line application

### PR DESCRIPTION
**Pull request by led02**

### CODE REVIEW

File: src/hermes/cli.py
Comment (sdruskat) [inline src/hermes/cli.py:26]:
```suggestion
        Overridden to ensure the order of commands is retained.
```

Think this is an override rather than a replacement?

File: src/hermes/cli.py
Comment (sdruskat) [inline src/hermes/cli.py:36]:
```suggestion
        Overridden to return commands in fixed order.
```

see above

**Review by sdruskat — COMMENTED** [review]
Thanks for this PR 🌮!
Flawless implementation and a clean 100% test coverage 🥇.

Two minor comments, and a more general question/request:

- Do we want to make sure that any PR also updates documentation if necessary? (If so, could reflect this in a PR template.)

E.g., for this PR I'd have expected the README to contain a wee section about *Usage* ("installing" + perhaps just the output of `haggis --help`/`python -m hermes --help`), and a wee section about *Running tests* or similar, including a pointer towards pytest-cov.

Happy to discuss this in a separate issue. This as a comment, as I'd like to have more eyes on this PR in this early stage of implementation, and will invite more reviews.

File: README.md
Comment (sdruskat) [inline README.md:41]:
```suggestion
[^1]: Working title, might be subject to change.
```

Fix typo.

File: pyproject.toml
Comment (sdruskat) [inline pyproject.toml:21]: N00b extension points question: This is just a dangling extension point for future implementation against, right?

**Review by sdruskat — APPROVED** [review]
Thanks for this PR again!

All in all, this is good to go IMHO, which is why I'm approving.

Albeit, I've made some suggestions in the files themselves 🙈 and would suggest the following changes, either before merge (e.g. while waiting for reviews by @knodel and @poikilotherm), or in future PRs.

- Add `.coverage` and `__pycache__` directories to `.gitignore`
- Add `docs/build/` and `docs/source/api/` to `.gitignore`

**Comment by sdruskat** [comment]
Perhaps we should also bring @jkelling into the reviewers' fold, at least for this foundational PR?

File: README.md
Comment (led02) [reply README.md:63]: At least, it's consistent... ;)

File: docs/source/conf.py
Comment (led02) [inline docs/source/conf.py:54]:
```suggestion
html_static_path = ['_static']

```

**Comment by sdruskat** [comment]
Thanks for the new changes @led02. While I review once more, could you briefly comment on https://github.com/hermes-hmc/workflow/pull/13#discussion_r871195126 please?

**Review by sdruskat — APPROVED** [review]
Nice one, this is now good to go from my end! 🌮

**Comment by jkelling** [comment]
I do not have anything to remark about the code. Looks good to me.

I do have a question about the intended uses of those four commands `harvest`, ´process`, `deposit`, `post`. Is there a written rationale somewhere about what each one is supposed to do later and why the stages are split the way they are? I am especially unclear about what `post` is for ... why do we want post-process meta data after we already had it deposited? Is this not someone else's problem at that point?

**Review by jkelling — APPROVED** [review]

**Comment by sdruskat** [comment]
FYI @led02: @poikilotherm and @knodel have agreed to have this merged.
In your absence, I'll do this now.

**Comment by led02** [comment]
Thanks!

Von meinem iPhone gesendet

**Pull request merged**
